com.unity.render-pipelines.universal/Editor/UniversalRenderPipelineLightEditor.cs

```diff
@@ -214,243 +171,121 @@ void SetOptions(AnimBool animBool, bool initialize, bool targetValue)
-            if (EditorGUI.EndChangeCheck())
+            // 0: Custom bias - 1: Bias values defined in Pipeline settings
+            int selectedUseAdditionalData = serializedLight.additionalLightData.usePipelineSettings ? 1 : 0;
+            Rect r = EditorGUILayout.GetControlRect(true);
+            EditorGUI.BeginProperty(r, Styles.shadowBias, serializedLight.useAdditionalDataProp);
             {
-                hasChanged = true;
-            }
-            if (m_AdditionalLightDataSO != null)
-                EditorGUI.EndProperty();
-
-            // Check mixed values
-            if (!m_UseAdditionalDataProp.hasMultipleDifferentValues)
-            {
-                if (selectedUseAdditionalData != 1 && m_AdditionalLightDataSO != null)
+                using (var checkScope = new EditorGUI.ChangeCheckScope())
                 {
-                    EditorGUI.indentLevel++;
-                    EditorGUILayout.Slider(settings.shadowsBias, 0f, 10f, "Depth");
-                    EditorGUILayout.Slider(settings.shadowsNormalBias, 0f, 10f, Styles.ShadowNormalBias);
-                    EditorGUI.indentLevel--;
+                    selectedUseAdditionalData = EditorGUI.IntPopup(r, Styles.shadowBias, selectedUseAdditionalData, Styles.displayedDefaultOptions, Styles.optionDefaultValues);
+                    if (checkScope.changed)
+                    {
+                        foreach (var additionData in serializedLight.lightsAdditionalData)
+                            additionData.usePipelineSettings = selectedUseAdditionalData != 0;
 
-                    m_AdditionalLightDataSO.ApplyModifiedProperties();
+                        serializedLight.Apply();
+                    }
                 }
             }
+            EditorGUI.EndProperty();
 
-            if (hasChanged)
+            if (!serializedLight.useAdditionalDataProp.hasMultipleDifferentValues)
             {
-                if (m_AdditionalLightDataSO == null)
+                if (selectedUseAdditionalData != 1) // Custom Bias
                 {
-                    MultipleAdditionalLightDataCheck();
-                    SetupSettings();
-                }
-                else
-                {
-                    foreach (var lightTarget in targets)
+                    using (new EditorGUI.IndentLevelScope())
                     {
-                        var additionData = (lightTarget as Component).gameObject.GetComponent<UniversalAdditionalLightData>();
-                        if (additionData == null)
-                            break;
-
-                        additionData.usePipelineSettings = selectedUseAdditionalData != 0;
+                        using (var checkScope = new EditorGUI.ChangeCheckScope())
+                        {
+                            EditorGUILayout.Slider(settings.shadowsBias, 0f, 10f, Styles.ShadowDepthBias);
```

Comment:
Sometimes you're using `settings` and other times `serializedLight.settings` - are these referring to the same data? Is there a difference or should refer to just one of these for the sake of consistency?

Reply:
They are the same data, and should point to `serializedLight.settings`. I will have a scond look, thank you :)